code/modules/research/nanites/nanite_programs/protocols.dm

```diff
@@ -98,3 +80,188 @@
+		allowing them to reach much higher saturation levels, but at the risk of causing internal damage to the host."
+	use_rate = 0
+	rogue_types = list(/datum/nanite_program/necrotic)
+	protocol_class = NANITE_PROTOCOL_STORAGE
+	var/extra_volume = 1500
+	var/next_warning = 0
+	var/min_warning_cooldown = 120
+	var/max_warning_cooldown = 350
+	var/volume_warnings_stage_1 = list("You feel a dull pain in your abdomen.",
+									"You feel a tickling sensation in your abdomen.")
+	var/volume_warnings_stage_2 = list("You feel a dull pain in your stomach.",
+									"You feel a dull pain when breathing.",
+									"Your stomach grumbles.",
+									"You feel a tickling sensation in your throat.",
+									"You feel a tickling sensation in your lungs.",
+									"You feel a tickling sensation in your stomach.",
+									"Your lungs feel stiff.")
+	var/volume_warnings_stage_3 = list("You feel a dull pain in your chest.",
+									"You hear a faint buzzing coming from nowhere.",
+									"You hear a faint buzzing inside your head.",
+									"Your head aches.")
+	var/volume_warnings_stage_4 = list("You feel a dull pain in your ears.",
+									"You feel a dull pain behind your eyes.",
+									"You hear a loud, echoing buzz inside your ears.",
+									"You feel dizzy.",
+									"You feel an itch coming from behind your eyes.",
+									"Your eardrums itch.",
+									"You see tiny grey motes drifting in your field of view.")
+	var/volume_warnings_stage_5 = list("You feel sick.",
+									"You feel a dull pain from every part of your body.",
+									"You feel nauseous.")
+	var/volume_warnings_stage_6 = list("Your skin itches and burns.",
+									"Your muscles ache.",
+									"You feel tired.",
+									"You feel something skittering under your skin.",)
+
+/datum/nanite_program/protocol/unsafe_storage/enable_passive_effect()
+	. = ..()
+	nanites.set_max_volume(null, nanites.max_nanites + extra_volume)
+
+/datum/nanite_program/protocol/unsafe_storage/disable_passive_effect()
+	. = ..()
+	nanites.set_max_volume(null, nanites.max_nanites - extra_volume)
+
+/datum/nanite_program/protocol/unsafe_storage/active_effect()
+	if(!iscarbon(host_mob))
+		if(prob(10))
+			host_mob.adjustBruteLoss(((max(nanites.nanite_volume - 450, 0) / 450) ** 2 ) * 0.5) // 0.5 -> 2 -> 4.5 -> 8 damage per successful tick
+		return
+
+	var/mob/living/carbon/C = host_mob
+	var/obj/item/organ/liver/liver = C.getorganslot(ORGAN_SLOT_LIVER)
+	var/obj/item/organ/stomach/stomach = C.getorganslot(ORGAN_SLOT_STOMACH)
+	var/obj/item/organ/lungs/lungs = C.getorganslot(ORGAN_SLOT_LUNGS)
+	var/obj/item/organ/heart/heart = C.getorganslot(ORGAN_SLOT_HEART)
+	var/obj/item/organ/eyes/eyes = C.getorganslot(ORGAN_SLOT_EYES)
+	var/obj/item/organ/ears/ears = C.getorganslot(ORGAN_SLOT_EARS)
+	var/obj/item/organ/brain/brain = C.getorganslot(ORGAN_SLOT_BRAIN)
+
+	if(nanites.nanite_volume < 500)
```

Comment:
This whole block just reallly sucks, is there some way you can make it better? a for loop maybe? Or is variation between each organ needed

Reply:
Can't really see a way around having each getorganslot separately. I could perhaps make a list of organ slots and use it to initialize a list of generic organs, but i suspect that it would obfuscate the code for no great gain.